llvm/lib/Target/AIE/AIEBaseInstrInfo.cpp

```diff
@@ -172,8 +180,11 @@ bool AIEBaseInstrInfo::analyzeBranch(MachineBasicBlock &MBB,
   if (NumTerminators == 1 && I->getDesc().isConditionalBranch()) {
     if (isHardwareLoopEnd(I->getOpcode())) {
       TBB = I->getOperand(0).getMBB();
+      // The machine verifier expects a non-empty Cond array, and we need to
+      // supply enough information to reconstruct this instruction.
+      // The basic block target is provided in the interfaces so we don't need
+      // to push that
       Cond.push_back(MachineOperand::CreateImm(I->getOpcode()));
-      Cond.push_back(I->getOperand(0));
```

Comment:
Did that hurt to keep `Cond.push_back(I->getOperand(0));`? I'm still struggling to understand what kind of API `analyzeBranch` has.

Reply:
The api is that the target can push whatever it needs to reconstruct/invert a branch. We don't actually need that third operand, and I like occam's razor.